xtraplatform-features/src/main/java/de/ii/xtraplatform/features/domain/FeaturePropertyV2.java

```diff
@@ -50,19 +55,29 @@ default Type getType() {
         return Type.STRING;
     }
 
+    Optional<String> getObjectType();
+
+    Optional<Type> getValueType();
+
     Optional<Role> getRole();
 
+    Optional<String> getLabel();
+
+    Optional<String> getDescription();
+
+    Optional<SimpleFeatureGeometry> getGeometryType();
+
+    Optional<Map<String, String>> getTransformers();
+
+    Map<String, List<String>> getConstraints();
```

Comment:
For this you should implemented a dedicated interface `Contraints` with the common annotations for immutable generation and deserialization. The line above would then be `Optional<Contraints> getContraints();`.
The interface should have getters for the possible keys like `codelist`, `enum` or `min` and use the proper types like `String`, `List<String>` or `double`.